### PR TITLE
opentelemetry: prepare for v0.5.0 release

### PR DESCRIPTION
**Pull request by jtescher**
Change log:

### Added

- Support `tracing-log` special values (#735)
- Support `Span::follows_from` creating otel span links (#723)
- Dynamic otel span names via `otel.name` field (#732)

### Breaking Changes

- Upgrade to `v0.6.0` of `opentelemetry` (#745)

### Fixed

- Filter out invalid parent contexts when building span contexts (#743)

### CODE REVIEW

File: tracing-opentelemetry/src/layer.rs [hunk 1]
```diff
@@ -102,7 +102,6 @@ pub(crate) fn build_span_context(
                 let mut result = sampler.should_sample(
                     builder.parent_context.as_ref(),
                     trace_id,
-                    span_id.clone(),
```
Comment (davidbarsky) [inline tracing-opentelemetry/src/layer.rs:105]: Why's this included in a release PR?

Comment (jtescher) [reply tracing-opentelemetry/src/layer.rs:105]: Hm yeah this was an impl change for the opentelemetry 0.6.0 upgrade, I can pull out to separate PR

Comment (jtescher) [reply tracing-opentelemetry/src/layer.rs:105]: 👍 Pulled these changes into https://github.com/tokio-rs/tracing/pull/745

File: tracing-opentelemetry/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,19 @@
+# 0.5.0 (June 2, 2020)
+
```
Comment (hawkw) [inline tracing-opentelemetry/CHANGELOG.md:2]: Can we provide some kind of list of the breaking changes or some upgrade notes?

File: tracing-opentelemetry/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "tracing-opentelemetry"
-version = "0.4.0"
+version = "0.5.0"
```
Comment (hawkw) [inline tracing-opentelemetry/Cargo.toml:3]: Let's make sure there are not also references to the prior version in the README and in RustDoc before merging this.

File: tracing-opentelemetry/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,19 @@
...
+- Support `Span::follows_from` creating otel span links
+- Dynamic otel span names via `otel.name` field
+
+### Changed
+
+- Upgrade to `v0.6.0` of `opentelemetry`
+
+### Fixed
+
+- Filter out invalid parent contexts when building span contexts
```
Comment (hawkw) [inline tracing-opentelemetry/CHANGELOG.md:15]: Usually, I like to reference PR numbers in changelog entries, can we do that here? Thanks!

File: tracing-opentelemetry/Cargo.toml (hunk 3 above)
Comment (jtescher) [reply tracing-opentelemetry/Cargo.toml:3]: Checked, shouldn't have any 0.4.0 in the docs 👍

File: tracing-opentelemetry/CHANGELOG.md (hunk 4 above)
Comment (jtescher) [reply tracing-opentelemetry/CHANGELOG.md]: added 👍

File: tracing-opentelemetry/CHANGELOG.md (hunk 2 above)
Comment (jtescher) [reply tracing-opentelemetry/CHANGELOG.md:2]: The small breaking change is in the `opentelemetry` dependency and should only be for users who are making custom samplers. I changed the section title to breaking changes so folks know they have to upgrade, think it's wroth documenting more here?

Comment (hawkw) [reply tracing-opentelemetry/CHANGELOG.md:2]: That's probably fine.

**Pull request merged**